### PR TITLE
Feature/impact hub multi ads

### PR DESCRIPTION
**Pull request by dodo721**
Bring in changes to allow impact hubs with multiple ads to display appropriately, along with new integrations for campaigns to hide low donation charities and/or specific charities.

### CODE REVIEW

File: src/style/CampaignPage.less [hunk 1]
```diff
@@ -215,8 +215,8 @@ body {
 
 	.theunit {
 		z-index: 1;
-		width: 83%;
-		top: 47%;
+		width: 57%;
+		top: 49.5%;
```
Comment (winterstein) [inline src/style/CampaignPage.less:219]: Doc - Maybe a note as to why these %s?

File: src/js/plumbing/ServerIO.js [hunk 2]
```diff
@@ -37,7 +37,7 @@ ServerIO.PROFILER_ENDPOINT = `${C.HTTPS}://${C.SERVER_TYPE}profiler.good-loop.co
 // ServerIO.AS_ENDPOINT = `${C.HTTPS}://${C.SERVER_TYPE}as.good-loop.com`;
 // ServerIO.AS_ENDPOINT = `https://testas.good-loop.com`;
 // Use the live adserver, since our showcase ad selectionis hard-coded to live ads.
-ServerIO.AS_ENDPOINT = 'https://as.good-loop.com';
+ServerIO.AS_ENDPOINT = 'http://as.good-loop.com';
```
Comment (winterstein) [inline src/js/plumbing/ServerIO.js:40]: Why http? Thanks. We normally want urls to be https, as some systems and browser requests demand that.

NB: Local urls are an exception 'cos local ssl is a pain to setup.

File: src/js/components/campaignpage/Charities.jsx [hunk 3]
```diff
@@ -43,11 +44,45 @@ challenges facing our planet."`,
...
+ * @param {?Number} lowDonationThreshold Custom threshold number to treat charities as "low donation". 1% of the total by default
+ * @param {?Boolean} showLowDonations If false will hide the donation number for low donation charities
+ * @param {?Boolean} showDonations If false will hide all donation numbers
+ * @param {?String[]} hideCharities hide specific charities by ID
  */
-const Charities = ({ charities, donation4charity }) => {
+const Charities = ({ charities, donation4charity, campaignPage }) => {
 	
+	//filterLowDonations, lowDonationThreshold, showLowDonations, showDonations, hideCharities
+	let {lowDntnDisplay, lowDntn, hideCharities} = campaignPage;
```
Comment (winterstein) [inline src/js/components/campaignpage/Charities.jsx:56]: Are these part of the CampaignPage object?

File: src/js/components/campaignpage/CampaignPage.jsx [hunk 4]
```diff
@@ -574,32 +578,29 @@ const AdvertCard = ({ ad }) => {
...
 	// Show when the campaign ran
 	// Fallback to ad creation date
-	const durationText = ad.start || ad.end ? <span>
-		{ad.start ? <Misc.RoughDate date={ad.start} /> : ''}
-		{ad.start && ad.end ? ' - ' : null}
-		{ad.end ? <Misc.RoughDate date={ad.end} /> : ''}
-	</span> : <span>
-		<Misc.RoughDate date={ad.created} />
-	</span>;
+	const durationText = <Misc.DateDuration startDate={ad.start} endDate={ad.end}/>;
```
Comment (winterstein) [inline src/js/components/campaignpage/CampaignPage.jsx:598]: (minor) Let's move `durationText` directly into the returned jsx.

**Review by winterstein — CHANGES_REQUESTED** [review]
Hello Ben

I'm unclear on the changes to the CampaignPage data-type. Possibly this is covered elsewhere, e.g. adserver.

Dan

File: src/js/components/campaignpage/Charities.jsx [hunk 5]
```diff
@@ -58,6 +93,13 @@ const Charities = ({ charities, donation4charity }) => {
 	};
 
 	let sogiveCharitiesWithDonations = sogiveCharities.filter(c => getDonation(c)); // Get rid of charities with no logged donations.
+	if (hideCharities) {
```
Comment (winterstein) [inline src/js/components/campaignpage/Charities.jsx:96]: What data format is hideCharities? Maybe rename it to make it clearer.

File: src/js/components/campaignpage/Charities.jsx (hunk 3 above)
Comment (dodo721) [reply src/js/components/campaignpage/Charities.jsx:56]: Yes - comments added

File: src/js/components/campaignpage/CampaignPage.jsx (hunk 4 above)
Comment (dodo721) [reply src/js/components/campaignpage/CampaignPage.jsx]: done

File: src/js/components/campaignpage/Charities.jsx (hunk 5 above)
Comment (dodo721) [reply src/js/components/campaignpage/Charities.jsx:111]: Added comments - it's represented by the KeySet prop control as an object of {charity id : bool}, which I then convert to an array

**Comment by dodo721** [comment]
Hi @winterstein ,
Made changes and added comments. Tried to explain the new data in the CampaignPage object in Charities.jsx where it's used.

**Review by winterstein — APPROVED** [review]
Thanks Ben :+1:

**Pull request merged**